script.js

```diff
@@ -1,4 +1,182 @@
+    if (!(input === "")) {
+      return `Hello ${callPlayer(
+        playerTurn
+      )}!<br>Do NOT type anything and submit to start the round!`;
+    }
+    rollDice();
+    gameState += 1;
+    return showOptions(recentRolls);
+  } else if (gameState === 1 || gameState === 3) {
+    if (!validP1P3(input)) {
+      return `ERROR 1. Please input ${totalDice} numbers ONLY! And they should not have any repeats! The numbers are ${recentRolls}`;
+    }
+    let splitNumbers = input.split("");
+    console.log(`splitNumbers = ${splitNumbers}`);
+    if (checkDupes(splitNumbers) === false) {
+      return `ERROR 2.Please input ${totalDice} numbers ONLY! And they should not have any repeats! The numbers are ${recentRolls}`;
+    }
+    if (checkNumbers(splitNumbers) === false) {
+      return `ERROR 3.Please input ${totalDice} numbers ONLY! And they should not have any repeats! The numbers are ${recentRolls}`;
+    }
+    if (playerTurn === 1) {
+      for (let i = 0; i < splitNumbers.length; i += 1) {
+        player1Hand += recentRolls[splitNumbers[i] - 1];
+      }
+    } else {
+      for (let i = 0; i < splitNumbers.length; i += 1) {
+        player2Hand += recentRolls[splitNumbers[i] - 1];
+      }
+    }
+    let message = "";
+    if (playerTurn === 1) {
+      message = `${callPlayer(
+        playerTurn
+      )} has opted for ${player1Hand}! Press Submit to proceed to Player 2's turn!`;
+    } else {
+      message = `${callPlayer(
+        playerTurn
+      )} has opted for ${player2Hand}! Press Submit to head towards the resolution!`;
+    }
+    if (playerTurn === 1) {
+      playerTurn = 2;
+    } else {
+      playerTurn = 1;
+    }
+    gameState += 1;
+    recentRolls = [];
+    return message;
+  } else {
+    gameState = 0;
+    winnerMessage = compareScores(player1Hand, player2Hand);
+    player1Hand = "";
+    player2Hand = "";
+    return winnerMessage;
+  }
+}
+
+// // differentiating between the player
+
+function callPlayer(playerTurn) {
+  if (playerTurn === 1) {
+    return "Player 1";
+  } else {
+    return "Player 2";
+  }
+}
+
+//roll2dice
+
+function rollDice() {
+  for (let i = 0; totalDice > i; i += 1) {
+    recentRolls.push(Math.ceil(Math.random() * 6));
+  }
+  console.log(recentRolls);
+  return;
+}
+// a function to show options
+function showOptions(playerRolls) {
+  return `Hello ${callPlayer(
+    playerTurn
+  )}! Your rolls are ${playerRolls}. Please key in the order you prefer the dice to go in! (Example: 2431 means 2nd dice, 4th dice, 3rd dice then 1st dice respectively!)`;
+}
+
+// checking duplicates
+function checkDupes(testsubject) {
+  for (let i = 0; i < testsubject.length; i++) {
+    for (let j = i + 1; j < testsubject.length; j++) {
+      if (testsubject[i] === testsubject[j]) {
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+//comparing scores and setting it to
+function compareScores(player1Hand, player2Hand) {
+  let winner = "";
+  Number(player1Hand);
```

Comment:
You made a similar mistake the last time during class. Rmb that casting Number(player1Hand) does not automatically make player1Hand a number. You have to store the result in a variable.
```js
var newHand = Number(player1Hand)
```